src/NLog/Targets/Wrappers/AsyncRequestQueueBase.cs

```diff
@@ -40,12 +40,18 @@ namespace NLog.Targets.Wrappers
 
     internal abstract class AsyncRequestQueueBase
     {
+        protected volatile int _requestLimit;
+
         public abstract bool IsEmpty { get; }
 
         /// <summary>
         /// Gets or sets the request limit.
         /// </summary>
-        public int RequestLimit { get; set; }
+        public int RequestLimit
```

Comment:
Please revert this back to simple property without volatile, since it gives a performance hit.